caddytest/caddytest.go

```diff
@@ -371,3 +392,75 @@ func AssertAdapt(t *testing.T, rawConfig string, adapterName string, expectedRes
 		t.Fail()
 	}
 }
+
+// AssertPostResponse asserts whether POST request returns expected status code and body.
+func AssertPostResponse(t *testing.T, requestURI string, requestHeaders []string, requestBody *bytes.Buffer, statusCode int, expectedBody string) (*http.Response, string) {
+	tester := NewTester(t)
+	return tester.AssertPostResponse(requestURI, requestHeaders, requestBody, statusCode, expectedBody)
+}
+
+// AssertPostResponse asserts whether POST request returns expected status code and body.
+func (tc *Tester) AssertPostResponse(requestURI string, requestHeaders []string, requestBody *bytes.Buffer, statusCode int, expectedBody string) (*http.Response, string) {
+	resp, body := AssertPostResponseBody(tc, requestURI, requestHeaders, requestBody, statusCode)
+	if !strings.Contains(body, expectedBody) {
+		tc.t.Errorf("requesting \"%s\" expected response body \"%s\" but got \"%s\"", requestURI, expectedBody, body)
+	}
+	return resp, body
+}
+
+// AssertPostResponseBody performs http call to assert whether POST request returns expected status code and body.
+func AssertPostResponseBody(tc *Tester, requestURI string, requestHeaders []string, requestBody *bytes.Buffer, expectedStatusCode int) (*http.Response, string) {
```

Comment:
I think we should push the requests through a common AssertStatusCode function that will take a http.Request object which will mean we can do other verbs too

https://github.com/caddyserver/caddy/pull/3285/files#diff-04cda1e6d4da174d1919b0c57a3c86f9R265

I am fine if want ot

Reply:
>> I think we should push the requests through a common AssertStatusCode function that will take a http.Request object which will mean we can do other verbs too

:+1: